presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -381,6 +383,42 @@ public static Long stNumInteriorRings(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return Long.valueOf(((OGCPolygon) geometry).numInteriorRing());
     }
 
+    @Description("Returns the cardinality of the geometry collection")
+    @ScalarFunction("ST_NumGeometries")
+    @SqlType(StandardTypes.INTEGER)
+    public static long stNumGeometries(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        if (geometry.isEmpty()) {
+            return 0;
+        }
+        GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
+        if (!type.isMultitype()) {
+            return 1;
+        }
+        return ((OGCGeometryCollection) geometry).numGeometries();
+    }
+
+    @SqlNullable
+    @Description("Returns the geometry element at the specified index, index started with 1")
+    @ScalarFunction("ST_GeometryN")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stGeometryN(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlType(INTEGER) long index)
+    {
+        OGCGeometry geometry = deserialize(input);
+        GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
+        if (!type.isMultitype()) {
+            if (index == 1) {
+                return input;
+            }
+            return null;
+        }
+        if (index < 1 || index > ((OGCGeometryCollection) geometry).numGeometries()) {
+            return null;
+        }
+        return serialize(((OGCGeometryCollection) geometry).geometryN((int) index - 1));
```

Comment:
This expression looks pretty complex. It will be more readable to split this up:

```
OGCGeometry ogcGeometry = ((OGCGeometryCollection) geometry).geometryN((int) index - 1);
return serialize(ogcGeometry);
```